frontend/cmd/cmd.go

```diff
@@ -101,21 +103,29 @@ func correlationIDPolicy(req *policy.Request) (*http.Response, error) {
 }
 
 func (opts *FrontendOpts) Run() error {
+	ctx := context.Background()
+
 	logger := util.DefaultLogger()
 	logger.Info(fmt.Sprintf("%s (%s) started", frontend.ProgramName, util.Version()))
 
-	// Init the Prometheus emitter.
+	// Initialize the Prometheus emitter.
 	prometheusEmitter := frontend.NewPrometheusEmitter(prometheus.DefaultRegisterer)
 
+	// Initialize the global OpenTelemetry tracer.
+	otelShutdown, err := frontend.ConfigureOpenTelemetryTracer(ctx, logger)
+	if err != nil {
+		return fmt.Errorf("could not initialize opentelemetry sdk: %w", err)
+	}
+
 	// Create the database client.
-	ctx := context.Background()
 	cosmosDatabaseClient, err := database.NewCosmosDatabaseClient(
 		opts.cosmosURL,
 		opts.cosmosName,
 		azcore.ClientOptions{
 			// FIXME Cloud should be determined by other means.
 			Cloud:           cloud.AzurePublic,
 			PerCallPolicies: []policy.Policy{policyFunc(correlationIDPolicy)},
+			TracingProvider: azotel.NewTracingProvider(otel.GetTracerProvider(), nil),
```

Comment:
Does the azotel trace provider something special?

Reply:
the goal here is to wrap the OTEL tracer provider into the Azure SDK tracer provider. In other words, the Azure SDK has its own definition of a tracer which is different (though close) from the OTEL SDK.

Reply:
Should we then also make use of it on cluster-service?

Reply:
Right I thought that I already created a ticket for this but apparently not.

https://issues.redhat.com/browse/ARO-15075